modules/analysis/src/model.rs

```diff
@@ -35,7 +37,35 @@ pub struct PackageNode {
     pub document_id: String,
     pub product_name: String,
     pub product_version: String,
+    pub approximate_memory_size: u32,
+}
+
+impl PackageNode {
+    pub(crate) fn set_approximate_memory_size(&self) -> PackageNode {
+        // Is there a better way to do this?
+        let size = size_of::<PackageNode>()
+            + self.sbom_id.len()
+            + self.node_id.len()
+            + self.purl.iter().fold(0, |acc, purl|
+                // use the json string length as an approximation of the memory size
+                acc + serde_json::to_string(purl).unwrap_or_else(|_| "".to_string()).len())
+            + self.cpe.iter().fold(0, |acc, cpe|
+                // use the json string length as an approximation of the memory size
+                acc + serde_json::to_string(cpe).unwrap_or_else(|_| "".to_string()).len())
+            + self.name.len()
+            + self.version.len()
+            + self.published.len()
+            + self.document_id.len()
+            + self.product_name.len()
+            + self.product_version.len();
+
+        PackageNode {
+            approximate_memory_size: size.try_into().unwrap_or(u32::MAX),
+            ..self.clone()
+        }
+    }
```

Comment:
Is there a better way to estimate how much memory this thing uses up?

Reply:
Kinda assuming that saving the approximate_memory_size will give us savings later, it might not.

Reply:
I switched to using the DeepSizeOf train which has a derive macro and it seems to give much better results.